torch/_inductor/codegen/cpp.py

```diff
@@ -1209,8 +1209,19 @@ def scale_index_with_offset(
         new_index = sympy_subs(index, replacement)
         return new_index
 
-    @staticmethod
-    def indirect_indexing(index_var, size, check=True):
+    def indirect_indexing(self, index_var, size, check=True):
+        generate_assert = (
+            (check or config.debug_index_asserts)
+            and config.cpp.assert_indirect_indexing
+        )
+
+        if generate_assert:
+            expr = ops.and_(ops.ge(index_var, '0'),
+                            ops.le(index_var, self.rename_indexing(size)))
+            cond = self.cse.generate(self.compute, expr)
+            line = f'assert({cond} && "index out of bounds");'
```

Comment:
Rather than doing a naked assert, please use `TORCH_CHECK`.

Reply:
Also, do you need to handle `indirect_max_sizes` the similar way as triton here?

Reply:
Yep, that's https://github.com/pytorch/pytorch/pull/108910#discussion_r1320610435. Note that that's an optional optimisation, but it'd be good to have now that we're at it.